MetricsUploader/MetricsUploaderWindows.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/memory/memory.h>
+
+#include "MetricsUploader/MetricsUploader.h"
+#include "OrbitBase/Logging.h"
+
+namespace orbit_metrics_uploader {
+
+constexpr const char* kSendLogEventFunctionName = "SendOrbitLogEvent";
+constexpr const char* kShutdownUploaderClientConnName = "ShutdownUploaderClientConn";
+constexpr const char* kStartUploaderClientFunctionName = "StartUploaderClient";
+
+MetricsUploader::~MetricsUploader() {
+  absl::MutexLock lock(&send_log_event_mutex_);
+  send_log_event_addr_ = nullptr;
+  HMODULE metrics_uploader_client_dll;
+  if (GetModuleHandleExA(GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT, client_name_.c_str(),
+                         &metrics_uploader_client_dll) != 0) {
+    FreeLibrary(metrics_uploader_client_dll);
+  }
+}
+
+ErrorMessageOr<std::unique_ptr<MetricsUploader>> MetricsUploader::CreateMetricsUploader(
+    std::string_view client_name) {
+  HMODULE metrics_uploader_client_dll;
+  if (GetModuleHandleExA(GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT, client_name.data(),
+                         &metrics_uploader_client_dll) != 0) {
+    return ErrorMessage("MetricsUploader is already created");
+  }
+
+  metrics_uploader_client_dll = LoadLibraryA(client_name.data());
+  if (NULL != metrics_uploader_client_dll) {
+    LOG("Metrcs uploader client library is loaded");
+
+    Result (*start_uploader_client_addr_)();
+    start_uploader_client_addr_ =
+        (Result(*)())GetProcAddress(metrics_uploader_client_dll, kStartUploaderClientFunctionName);
+    if (NULL == start_uploader_client_addr_) {
+      std::string error_message =
+          absl::StrFormat("%s function not found", kStartUploaderClientFunctionName);
+      ERROR("%s", error_message);
+      FreeLibrary(metrics_uploader_client_dll);
```

Comment:
You probably should do this in the other error branches below as well, don't you?